src/components/AttachmentModal.js

```diff
@@ -326,6 +348,32 @@ function AttachmentModal(props) {
 
     const sourceForAttachmentView = props.source || source;
 
+    const threeDotMenuItems = [
+        {
+            icon: Expensicons.Camera,
+            text: props.translate('common.replace'),
+            onSelected: () => {
+                onModalHideCallbackRef.current = () => Navigation.navigate(ROUTES.EDIT_REQUEST.getRoute(props.report.reportID, CONST.EDIT_REQUEST_FIELD.RECEIPT));
+                closeModal();
+            },
+        },
+        {
+            icon: Expensicons.Download,
+            text: props.translate('common.download'),
+            onSelected: () => downloadAttachment(source),
+        },
+    ];
+
+    if (TransactionUtils.hasReceipt(props.transaction) && !TransactionUtils.isReceiptBeingScanned(props.transaction)) {
```

Comment:
I think `isReceiptBeingScanned` directly accesses `transaction.receipt.state` so maybe we shouldn't default the transaction to `undefined` in default props?

Reply:
We have to default it to something, right? Should we just add a sanity check here to make sure it is set?